src/main/kotlin/bsu/cc/schedule/ConflictChecking.kt

```diff
@@ -22,6 +23,23 @@ fun buildScheduleIntervalTree(): IntervalTree {
     return tree
 }
 
+fun checkConstraints(classes: Collection<ClassSchedule>,
+                     constraints: Collection<ClassConstraint>): Map<ClassConstraint, Set<List<ClassSchedule>>> {
+    val conflicts = HashMap<ClassConstraint, MutableList<ClassSchedule>>()
+    constraints.forEach{ conflicts[it] = ArrayList() }
+
+    classes.forEach { c ->
+        val classString = c.classString
+        constraints.forEach { if (it.classes.contains(classString)) conflicts[it]!!.add(c) }
```

Comment:
Can this be done with the looping reversed and a filter to avoid needing to use `!!`? Something like:
`contraints.forEach{ constraint ->
     conflicts[constraint] = classes.filter(contraints.classes.contains(it.classString)).toList() 
}`